onnxruntime/test/providers/cpu/controlflow/if_test.cc

```diff
@@ -339,5 +339,93 @@ TEST(If, Opset11ThenAndElseBranchesProduceDifferentOutputShapes) {
   RunTest(false, options, false, OpTester::ExpectResult::kExpectSuccess, "", 11);
 }
 
+// This is to test an "If" node with just "Constant" nodes in the "then" and "else" conditional branches
+class IfOpTesterOnlyConstantNodesInConditionalBranches : public OpTester {
+ public:
+  IfOpTesterOnlyConstantNodesInConditionalBranches() : OpTester("If") {
+  }
+
+ protected:
+  void AddNodes(onnxruntime::Graph& graph,
+                std::vector<onnxruntime::NodeArg*>& graph_input_defs,
+                std::vector<onnxruntime::NodeArg*>& graph_output_defs,
+                std::vector<std::function<void(onnxruntime::Node& node)>>& /*add_attribute_funcs*/) override {
+    // Graph inputs are 0:Cond for If
+    ASSERT_EQ(graph_input_defs.size(), 1u);
+    ASSERT_EQ(graph_output_defs.size(), 1u);
+
+    NodeArg* if_cond_input = graph_input_defs[0];
+
+    std::vector<NodeArg*> inputs;
+    std::vector<NodeArg*> outputs;
+
+    // add If node
+    {
+      inputs = {if_cond_input};
+      outputs = {graph_output_defs[0]};
+
+      auto& if_node = graph.AddNode("if", "If", "If node", inputs, outputs);
+
+      // Then branch - it has "Constant" node only
+      Model model_then("Then", false, DefaultLoggingManager().DefaultLogger());
```

Comment:
Could just have a lambda to create each subgraph that takes the value to use and returns a GraphProto

Reply:
Fixed